### PR TITLE
to compile on GCP and Azure

### LINKED ISSUE #952: Check PW_CSP and compile on GCP and Azure
After been able to compile on AWS, we need to make sure UFS_UTILS compile on Google (GCP) and Azure as well.

### PR DESCRIPTION
**Pull request by weihuang-jedi**
## DESCRIPTION OF CHANGES: 
Enable UFS_UTILS compilation on GCP and Azure.

## TESTS CONDUCTED: 

- [x] Compiled branch on GCP
- [x] Compiled branch on Azure
- [x] Compiled branch on AWS
- [x] Compiled branch at 472bbe1 on Hercules, Hera, Jet and WCOSS2. Did not compile on Orion as that machine was recently upgraded to Rocky 9 and UFS_UTILS has not been ported there yet. 

## DEPENDENCIES:
None

## ISSUE: 
Fixes #952.

### CODE REVIEW

File: sorc/machine-setup.sh [hunk 1]
```diff
@@ -38,17 +38,43 @@ elif [[ -d /scratch1 ]] ; then
     fi
     target=hera
     module purge
```
Comment (GeorgeGayno-NOAA) [inline sorc/machine-setup.sh]: These updates are for Gaea? Did you compile on Gaea? I don't have access to that machine, so you will need to check.

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
Please see my comment about Gaea.

File: sorc/machine-setup.sh (hunk 1 above)
Comment (weihuang-jedi) [reply sorc/machine-setup.sh]: George, 
Thanks for pointing out this, I did not write those code, those code somehow got in at the time I tried to sync my branch with develop, and messed up at my end.
I have re-sync with develop, and only added the code needed for cloud machines (AWS, Azure, GCP),
and compiled the new branch on AWS, GCP, and Azure, also compiled on Hera, and Gaea, all compiled fined.
Could please check again?
Thank you,
Wei

File: sorc/machine-setup.sh [hunk 2]
```diff
@@ -67,16 +67,22 @@ elif [[ -d /data/prod ]] ; then
     fi
     target=s4
     module purge
-elif [[ "$(dnsdomainname)" =~ "pw" ]]; then
-    if [[ "${PW_CSP}" == "aws" ]]; then # TODO: Add other CSPs here.
-	target=noaacloud
-        module purge
+else
+    if [[ ! -v PW_CSP ]]; then
```
Comment (GeorgeGayno-NOAA) [inline sorc/machine-setup.sh:71]: Not sure why you have two warning messages here.

File: sorc/machine-setup.sh [hunk 3]
```diff
@@ -67,16 +67,22 @@ elif [[ -d /data/prod ]] ; then
...
+        echo WARNING: UNSUPPORTED CSP PLATFORM 1>&2
+        echo WARNING: UNKNOWN PLATFORM 1>&2; exit 99
+    elif [[ -z "${PW_CSP}" ]]; then
+        echo WARNING: UNSUPPORTED CSP PLATFORM 1>&2
+        echo WARNING: UNKNOWN PLATFORM 1>&2; exit 99
     else
-        echo WARNING: UNSUPPORTED CSP PLATFORM 1>&2; exit 99
+        if [[ "${PW_CSP}" == "aws" || "${PW_CSP}" == "azure" || "${PW_CSP}" == "google" ]]; then
+            target=noaacloud
+            module purge
```
Comment (GeorgeGayno-NOAA) [inline sorc/machine-setup.sh:78]: On Hera, I tried to simulate an error on a non-supported, non-cloud platform:

```
 elif [[ -d /lfs/h1 ]] ; then
     target=wcoss2
     module reset
-elif [[ -d /scratch1 ]] ; then
+elif [[ -d /sratch1 ]] ; then
     # We are on NOAA Hera
```

I got two warnings - one for a CSP platform. 

```
++ echo WARNING: UNSUPPORTED CSP PLATFORM
WARNING: UNSUPPORTED CSP PLATFORM
++ echo WARNING: UNKNOWN PLATFORM
WARNING: UNKNOWN PLATFORM
```

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Comment by weihuang-jedi** [comment]
@GeorgeGayno-NOAA Do we need more reviewers, or we can merge this PR?
Thanks.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA Do we need more reviewers, or we can merge this PR? Thanks.

Merge the latest updates from 'develop'. Then I will merge this PR.

**Comment by weihuang-jedi** [comment]
I have sync my branch with "develop". Thanks.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Comment by GeorgeGayno-NOAA** [comment]
@weihuang-jedi - Thanks. Will merge.

**Comment by weihuang-jedi** [comment]
Thank you @George Gayno - NOAA Affiliate ***@***.***> !

**Pull request merged**